geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloIndexAdapter.scala

```diff
@@ -155,29 +156,71 @@ class AccumuloIndexAdapter(ds: AccumuloDataStore) extends IndexAdapter[AccumuloD
       val (cf, s) = groups.group(index.sft, hints.getTransformDefinition, ecql)
       (Some(new Text(AccumuloIndexAdapter.mapColumnFamily(index)(cf))), s)
     }
+    // used when remote processing is disabled
+    lazy val returnSchema = hints.getTransformSchema.getOrElse(schema)
+    lazy val fti = FilterTransformIterator.configure(schema, index, ecql, hints).toSeq
+    lazy val resultsToFeatures = AccumuloResultsToFeatures(index, returnSchema)
+    lazy val localReducer = {
+      val arrowHook = Some(ArrowDictionaryHook(ds.stats, filter.filter))
+      Some(new LocalTransformReducer(returnSchema, None, None, None, hints, arrowHook))
+    }
 
     index match {
       case i: AccumuloJoinIndex =>
         i.createQueryPlan(filter, tables, ranges, colFamily, schema, ecql, hints, numThreads)
 
       case _ =>
         val (iter, eToF, reduce) = if (strategy.hints.isBinQuery) {
-          val iter = BinAggregatingIterator.configure(schema, index, ecql, hints)
-          (Seq(iter), new AccumuloBinResultsToFeatures(), None)
+          if (ds.config.remote.bin) {
+            val iter = BinAggregatingIterator.configure(schema, index, ecql, hints)
+            (Seq(iter), new AccumuloBinResultsToFeatures(), None)
+          } else {
+            if (hints.isSkipReduce) {
+              // override the return sft to reflect what we're actually returning,
+              // since the bin sft is only created in the local reduce step
+              hints.hints.put(QueryHints.Internal.RETURN_SFT, returnSchema)
+            }
+            (fti, resultsToFeatures, localReducer)
```

Comment:
Above it looks like the localReducer gets an 'arrowHook'.  Does that get used / matter here?  

Or is it just expedient to express things that way?

Reply:
it's needed in certain query patterns to pre-calculate the dictionaries for an arrow query - it will get used if it's needed but otherwise not. it's basically just a reference back to the full filter for the query and the data store stats for calculating enumerations, which aren't otherwise available down where they're needed.